Sources/StreamChat/Workers/MessageUpdater.swift

```diff
@@ -255,15 +257,47 @@ class MessageUpdater: Worker {
         messageId: MessageId,
         completion: ((Error?) -> Void)? = nil
     ) {
+        var reaction: MessageReactionDTO?
+        let version = UUID().uuidString
+
         let endpoint: Endpoint<EmptyResponse> = .addReaction(
             type,
             score: score,
             enforceUnique: enforceUnique,
             extraData: extraData,
             messageId: messageId
         )
-        apiClient.request(endpoint: endpoint) {
-            completion?($0.error)
+
+        database.write { session in
+            do {
+                reaction = try session.addReaction(
+                    to: messageId,
+                    type: type,
+                    score: score,
+                    extraData: extraData
+                )
+            } catch {
+                log.warning("Failed to add the reaction to the database: \(error)")
+            }
+
+            if let reaction = reaction {
+                reaction.localState = .sending
+                reaction.version = version
+            }
+        } completion: { error in
+            self.apiClient.request(endpoint: endpoint, retryOptions: self.retryOptions) { result in
```

Comment:
If we've got the error from DB should we forward it to the caller and skip further steps? 🤔

Reply:
this is on purpose (DB fails to do the optimistic bit? ignore and move with API call)